src/liblsquic/lsquic_enc_sess_ietf.c

```diff
@@ -1973,7 +1973,7 @@ iquic_esf_encrypt_packet (enc_session_t *enc_session_p,
     dst_sz = lconn->cn_pf->pf_packout_size(lconn, packet_out);
     ipv6 = NP_IS_IPv6(packet_out->po_path);
     dst = enpub->enp_pmi->pmi_allocate(enpub->enp_pmi_ctx,
-                                packet_out->po_path->np_peer_ctx, dst_sz, ipv6);
+                                packet_out->po_path->np_peer_ctx, lsquic_conn_get_ctx(lconn), dst_sz, ipv6);
```

Comment:
This function is defined as follows:

```c
lsquic_conn_ctx_t *
lsquic_conn_get_ctx (const struct lsquic_conn *lconn)
{   
    return lconn->cn_if->ci_get_ctx(lconn);
}   
```

This means that for each call to `pmi_allocate()`, you call two more functions just to get the context.

A more difficult problem is that in server mode, the connection context does not exist until handshake completes.  (This is because none of the per-connection user callbacks have been called yet.)  Calling `lsquic_conn_get_ctx()` should crash because this method is not set in `mini_conn_ietf_iface`.  Did you test it in server mode?  It should not have worked.

Reply:
oh I see, I didn't investigate that it could be called directly. I'll change that.

and right I assumed in the pre-connection phase you'd handle a NULL connection by using malloc.

Reply:
With regard to the function call, I was thinking that maybe we can store that connection context pointer in `struct lsquic_conn` -- that would make it very cheap to access!  But then I realized that the connection context is not even defined before handshake is complete.  The actual call to `lsquic_conn_get_ctx()` will perform a NULL dereference if called by a mini conn.

Reply:
okay then seems best thing to do is store it in struct lsquic_conn, and then document that an implementor can expect to context to be NULL pre-handshake?

Reply:
Yes, you can give it a shot.  (Even though we went to great pain to minimize the size of `struct lsquic_conn`, so giving up eight whole bytes pains me!)

You should test your changes in server mode to make sure that the PR actually addresses your concern.  Otherwise, we may end up having to modify this again in the future...  We should try to get it right the first time.